sympy/utilities/_compilation/util.py

```diff
@@ -19,20 +19,7 @@ def may_xfail(func):
         return func
 
 
-if sys.version_info[0] == 2:
-    class FileNotFoundError(IOError):
-        pass
-
-    class TemporaryDirectory:
-        def __init__(self):
-            self.path = tempfile.mkdtemp()
-        def __enter__(self):
-            return self.path
-        def __exit__(self, exc, value, tb):
-            shutil.rmtree(self.path)
-else:
-    FileNotFoundError = FileNotFoundError
-    TemporaryDirectory = tempfile.TemporaryDirectory
+TemporaryDirectory = tempfile.TemporaryDirectory
```

Comment:
It would be better to remove this and import directly from tempfile wherever this is used.

Reply:
I did that in 2nd PR (https://github.com/sympy/sympy/pull/19875)

Reply:
Yes, I just saw that!

Reply:
It would probably be better to combine those changes in one PR (just push another commit to this branch) since the changes are small, related and overlapping.

Otherwise if either of these is merged then the other will have merge conflicts.